cgroup.go

```diff
@@ -30,24 +30,41 @@ import (
 )
 
 // New returns a new control via the cgroup cgroups interface
-func New(hierarchy Hierarchy, path Path, resources *specs.LinuxResources) (Cgroup, error) {
+func New(hierarchy Hierarchy, path Path, resources *specs.LinuxResources, skip ...SkipOpts) (Cgroup, error) {
 	subsystems, err := hierarchy()
 	if err != nil {
 		return nil, err
 	}
+	// if we don't have any skip opts, set the default to require the devices subsystem
+	if len(skip) == 0 {
+		skip = append(skip, RequireDevices)
+	}
+	var active []Subsystem
 	for _, s := range subsystems {
+		// check if subsystem exists
 		if err := initializeSubsystem(s, path, resources); err != nil {
+			if err == ErrControllerNotActive {
+				for _, o := range skip {
+					if skerr := o(s, path, err); skerr != nil {
+						if skerr != ErrSkipSubsystem {
+							return nil, skerr
+						}
+					}
+				}
+				continue
+			}
 			return nil, err
 		}
+		active = append(active, s)
 	}
 	return &cgroup{
 		path:       path,
-		subsystems: subsystems,
+		subsystems: active,
```

Comment:
how about adding a list for skipped .. or inactive?

Reply:
not sure we need it